library/std/src/io/impls.rs

```diff
@@ -410,3 +411,55 @@ impl<A: Allocator> Write for Vec<u8, A> {
         Ok(())
     }
 }
+
+/// Read is implemented for `VecDeque<u8>` by consuming bytes from the front of the `VecDeque`.
+#[stable(feature = "vecdeque_read_write", since = "1.62.0")]
+impl<A: Allocator> Read for VecDeque<u8, A> {
+    /// Fill `buf` with the contents of the "front" slice as returned by
+    /// [`as_slices`][`VecDeque::as_slices`]. If the contained byte slices of the `VecDeque` are
+    /// discontiguous, multiple calls to `read` will be needed to read the entire content.
+    #[inline]
+    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+        let (ref mut front, _) = self.as_slices();
+        let n = Read::read(front, buf)?;
+        self.drain(..n);
+        Ok(n)
+    }
+}
+
+/// Write is implemented for `VecDeque<u8>` by appending to the `VecDeque`.
+/// The `VecDeque` will grow as needed.
+#[stable(feature = "vecdeque_read_write", since = "1.62.0")]
+impl<A: Allocator> Write for VecDeque<u8, A> {
+    #[inline]
+    /// Append bytes to the `VecDeque`, only filling up to the available free space. Only a full
+    /// `VecDeque` (where [`len`][VecDeque::len] == [`capacity`][VecDeque::capacity]) will expand
+    /// its allocation to additionally hold `buf`.
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        let remaining = self.capacity() - self.len();
+
+        let n = match remaining {
+            0 => {
+                // Reserve space for the new bytes.
+                self.reserve(buf.len());
+                buf.len()
+            }
+            r => buf.len().min(r),
+        };
+
+        self.extend(&buf[..n]);
+        Ok(n)
+    }
+
+    #[inline]
+    fn write_all(&mut self, buf: &[u8]) -> io::Result<()> {
+        self.reserve(buf.len());
```

Comment:
This line shouldn't be necessary anymore after #95904 lands

Reply:
nice! I suppose whichever PR gets merged second can update this line?

Reply:
Done in #97922